### DISCUSSION TITLE
Siwave method on Linux system?

### DISCUSSION THREAD

**Discussion by terryli1980** [ansys/pyaedt#4242] [Q&A] [answered]
The method of using Siwave was asked before, as follows. Can this method be used on a Linux system?

```
from pyaedt.siwave import Siwave

siwave = Siwave(specified_version="2024.1")
siwave.open_project("xxx.siw")
siwave.oproject.ScrRunDcSimulation(1)

````

**Comment by Samuelopez-ansys** [comment]
@ring630 @svandenb-dev @terryli1980 

SIWave API method can not be used. This is PyAEDT, and there is not SIwave equivalent. BUT, there are some method in PyEDB that are calling SIwave API launching a script.

You can use PyAEDT directly, if you can solve your layout in HFSS 3DLayout, like in this example:
https://aedt.docs.pyansys.com/version/stable/examples/01-HFSS3DLayout/Dcir_in_3DLayout.html#sphx-glr-examples-01-hfss3dlayout-dcir-in-3dlayout-py

If you need SIwave, then you will need a similar approach than this one:
https://edb.docs.pyansys.com/version/stable/examples/legacy_standalone/01_edb_example.html#sphx-glr-examples-legacy-standalone-01-edb-example-py

If you prefer the EDB workflow, please transfer this discussion into the PyEDB repo:
https://github.com/ansys/pyedb

**Comment by hui-zhou-a** [comment, marked as answer]
@terryli1980 , PyEDB/PyAEDT has limited support for SIwave. If you want to run DC analysis, you can import Edb into HFSS 3DLayout. If you have existing siwave project, you can export it as edb and import it into HFSS 3DLayout.
